bokeh/core/has_props.py

```diff
@@ -605,10 +605,16 @@ def apply_theme(self, property_values):
         elif hasattr(self, '__themed_values__'):
             delattr(self, '__themed_values__')
 
+        # Property container values might be cached even if unmodified. Invalidate
+        # any cached values that are not modified at this point.
+        for k, v in old_values.items():
+            if isinstance(v, PropertyValueContainer) and k in self._unstable_default_values:
+                del self._unstable_default_values[k]
```

Comment:
If `self._unstable_themed_values` were separate, could reduce this loop to `self._unstable_themed_values = dict()`, but of course it will add code in other places that would now have to check three dicts (_unstable_themed_values, _unstable_default_values, and _property_values).